File: docker_compiler/js/executor.js
'use strict';

const fs = require('node:fs');
const perf_hooks = require('node:perf_hooks');
const { join } = require('node:path');

const RESULT_FILE_PATH = join('result', 'result.txt');

const solver = perf_hooks.performance.timerify(require('./solution.js'));

const performanceObserver = new perf_hooks.PerformanceObserver((items, observer) => {
	const entry = items.getEntriesByType('function').pop();
	fs.appendFileSync(RESULT_FILE_PATH, entry.duration.toString());
	observer.disconnect();
});
performanceObserver.observe({ entryTypes: ['function'] });

const inputData = fs.readFileSync('testcasesInputs.json', 'utf-8');
const outputData = fs.readFileSync('testcasesOutputs.json', 'utf-8');

const inputs = JSON.parse(inputData);
const outputs = JSON.parse(outputData);

const validate = (inputs, outputs) => {
	const inval = Object.values(inputs);
	const outval = Object.values(outputs);
	for (let i = 0; i < inval.length; i++) {
		const result = solver(inval[i]);
		if (result?.toString() !== outval[i].toString()) {
			const data = `Incorrect\nInput: ${inval[i]} | Expected output: ${outval[i]}\n`;
			fs.writeFileSync(RESULT_FILE_PATH, data);
			return;
		}
	}
	fs.writeFileSync(RESULT_FILE_PATH, 'Correct\n\n');
	return;
};
validate(inputs, outputs);

File: docker_compiler/py/executor.py
import json
import os
from time import perf_counter_ns

RESULT_FILE_PATH = os.path.join('result', 'result.txt')
solver = __import__('solution').solver

with open('testcasesInputs.json', 'r') as f:
    inputs = json.load(f)

with open('testcasesOutputs.json', 'r') as f:
    outputs = json.load(f)


def validate(inputs, outputs):
    inval = list(inputs.values())
    outval = list(outputs.values())
    for i in range(len(inval)):
        start_time = perf_counter_ns()
        result = solver(inval[i])
        end_time = perf_counter_ns()
        time_in_ms = (end_time - start_time) / 1000000.0
        if str([result]) != str(outval[i]):  # true != [true]
            print(str(result))
            print(str(outval[i]))
            data = f"Wrong answer - Input: {inval[i]} | Expected output: {outval[i]}\n{time_in_ms}"
            with open(RESULT_FILE_PATH, 'w') as f:
                f.write(data)
            return
    with open(RESULT_FILE_PATH, 'w') as f:
        f.write(f"Accepted\n{time_in_ms}")
    return


validate(inputs, outputs)
